### ISSUE TITLE
loss not changed.why?

### ISSUE THREAD

**Issue by ats4869** [alsgkals2/CoReD#7]
![image](https://github.com/user-attachments/assets/cf44bda7-779c-4133-b254-99b302f579f8)

**Comment by alsgkals2** [comment]
Hello,
in my case, the model works well.
Please check if the model is not set to eval mode and if all the data has been loaded correctly. If the issue occurred still, kindly send me the execution code and command line. Thank you.

**Comment by ats4869** [comment]
Thank you for your reply,
command line:
python main.py -t=DeepFake -d=/mnt/data1/zhangyufei/dataset/FFcored/
python main.py -s=Face2Face_DeepFake -t=FaceSwap -d=/mnt/data1/zhangyufei/dataset/FFcored/ -w=/mnt/data/zhangyufei/deepfake/CoReD/CoReD_weights
The loss of my training with the above two commands did not change.

I used the FF++ dataset, and my dataset format is ./DeepFake/train/fake/. ./DeepFake/train/real/. The problems I observed are: 1. The weight parameters of the model did not change during training. I increased lr to 100 and it changed. But loss and top did not change. My observation is that the outputs in outputs = student_model(inputs) predict the model as all 0 or all 1. I have tried normalizing the gradient and expanding the gradient. Changing transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) has no effect.


output：
[[ 0.0022, -0.0015],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0014],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0016],
        [ 0.0022, -0.0015],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0015],
        [ 0.0021, -0.0017],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0014],
        [ 0.0020, -0.0016],
        [ 0.0022, -0.0015],
        


        [ 0.0022, -0.0015],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0014],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0016],
        [ 0.0022, -0.0015],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0015],
        [ 0.0021, -0.0017],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0014],
        [ 0.0020, -0.0016],

        [ 0.0038, -0.0031],
        [ 0.0038, -0.0032],
        [ 0.0039, -0.0034],
        [ 0.0038, -0.0031],
        [ 0.0038, -0.0032],
        [ 0.0038, -0.0033],
        [ 0.0039, -0.0032],
        [ 0.0038, -0.0033],
        [ 0.0038, -0.0031],
        [ 0.0039, -0.0034],
        [ 0.0038, -0.0032],
        [ 0.0038, -0.0030],
        [ 0.0038, -0.0032],
        [ 0.0038, -0.0029],
        [ 0.0040, -0.0031],
        [ 0.0037, -0.0031],
        [ 0.0038, -0.0031],
        [ 0.0038, -0.0031],
        [ 0.0038, -0.0031],
        [ 0.0037, -0.0029],
        [ 0.0039, -0.0032],
![image](https://github.com/user-attachments/assets/5f49d581-0905-4514-9997-030c33a61650)
![image](https://github.com/user-attachments/assets/7c52f7f3-f6e2-4639-acc0-3fe692948b54)
I guess maybe scaler.scale(loss).backward() make grad too small to updata weight

**Comment by ats4869** [comment]
Thank you for your reply,
command line:
python main.py -t=DeepFake -d=/mnt/data1/zhangyufei/dataset/FFcored/
python main.py -s=Face2Face_DeepFake -t=FaceSwap -d=/mnt/data1/zhangyufei/dataset/FFcored/ -w=/mnt/data/zhangyufei/deepfake/CoReD/CoReD_weights
The loss of my training with the above two commands did not change.

I used the FF++ dataset, and my dataset format is ./DeepFake/train/fake/. ./DeepFake/train/real/. The problems I observed are: 1. The weight parameters of the model did not change during training. I increased lr to 100 and it changed. But loss and top did not change. My observation is that the outputs in outputs = student_model(inputs) predict the model as all 0 or all 1. I have tried normalizing the gradient and expanding the gradient. Changing transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) has no effect.


output：
[[ 0.0022, -0.0015],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0014],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0016],
        [ 0.0022, -0.0015],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0015],
        [ 0.0021, -0.0017],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0014],
        [ 0.0020, -0.0016],
        [ 0.0022, -0.0015],
        


        [ 0.0022, -0.0015],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0014],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0016],
        [ 0.0022, -0.0015],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0015],
        [ 0.0021, -0.0017],
        [ 0.0022, -0.0014],
        [ 0.0021, -0.0014],
        [ 0.0020, -0.0016],

        [ 0.0038, -0.0031],
        [ 0.0038, -0.0032],
        [ 0.0039, -0.0034],
        [ 0.0038, -0.0031],
        [ 0.0038, -0.0032],
        [ 0.0038, -0.0033],
        [ 0.0039, -0.0032],
        [ 0.0038, -0.0033],
        [ 0.0038, -0.0031],
        [ 0.0039, -0.0034],
        [ 0.0038, -0.0032],
        [ 0.0038, -0.0030],
        [ 0.0038, -0.0032],
        [ 0.0038, -0.0029],
        [ 0.0040, -0.0031],
        [ 0.0037, -0.0031],
        [ 0.0038, -0.0031],
        [ 0.0038, -0.0031],
        [ 0.0038, -0.0031],
        [ 0.0037, -0.0029],
        [ 0.0039, -0.0032],
![image](https://github.com/user-attachments/assets/5f49d581-0905-4514-9997-030c33a61650)
![image](https://github.com/user-attachments/assets/7c52f7f3-f6e2-4639-acc0-3fe692948b54)
I guess maybe scaler.scale(loss).backward()

**Comment by ats4869** [comment]
![image](https://github.com/user-attachments/assets/9508f519-a18e-40ef-924c-d34ceafb2604)
model.train

**Comment by ats4869** [comment]
![image](https://github.com/user-attachments/assets/3021b72b-bb20-4ab6-ad24-9e384a4429f2)
is there differences in --model and --network? xception not need pretrain weight but Efficient need pretrain weight ?
![image](https://github.com/user-attachments/assets/21e58d50-99b0-41a7-a3bf-65e3298b7163)

but loss not change yet

**Comment by alsgkals2** [comment]
> Thank you for your reply, command line: python main.py -t=DeepFake -d=/mnt/data1/zhangyufei/dataset/FFcored/ python main.py -s=Face2Face_DeepFake -t=FaceSwap -d=/mnt/data1/zhangyufei/dataset/FFcored/ -w=/mnt/data/zhangyufei/deepfake/CoReD/CoReD_weights The loss of my training with the above two commands did not change.
> 
> I used the FF++ dataset, and my dataset format is ./DeepFake/train/fake/. ./DeepFake/train/real/. The problems I observed are: 1. The weight parameters of the model did not change during training. I increased lr to 100 and it changed. But loss and top did not change. My observation is that the outputs in outputs = student_model(inputs) predict the model as all 0 or all 1. I have tried normalizing the gradient and expanding the gradient. Changing transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) has no effect.
> 
> output： [[ 0.0022, -0.0015], [ 0.0022, -0.0014], [ 0.0021, -0.0014], [ 0.0022, -0.0014], [ 0.0021, -0.0016], [ 0.0022, -0.0015], [ 0.0022, -0.0014], [ 0.0021, -0.0015], [ 0.0021, -0.0017], [ 0.0022, -0.0014], [ 0.0021, -0.0014], [ 0.0020, -0.0016], [ 0.0022, -0.0015],
> 
> ```
>     [ 0.0022, -0.0015],
>     [ 0.0022, -0.0014],
>     [ 0.0021, -0.0014],
>     [ 0.0022, -0.0014],
>     [ 0.0021, -0.0016],
>     [ 0.0022, -0.0015],
>     [ 0.0022, -0.0014],
>     [ 0.0021, -0.0015],
>     [ 0.0021, -0.0017],
>     [ 0.0022, -0.0014],
>     [ 0.0021, -0.0014],
>     [ 0.0020, -0.0016],
> 
>     [ 0.0038, -0.0031],
>     [ 0.0038, -0.0032],
>     [ 0.0039, -0.0034],
>     [ 0.0038, -0.0031],
>     [ 0.0038, -0.0032],
>     [ 0.0038, -0.0033],
>     [ 0.0039, -0.0032],
>     [ 0.0038, -0.0033],
>     [ 0.0038, -0.0031],
>     [ 0.0039, -0.0034],
>     [ 0.0038, -0.0032],
>     [ 0.0038, -0.0030],
>     [ 0.0038, -0.0032],
>     [ 0.0038, -0.0029],
>     [ 0.0040, -0.0031],
>     [ 0.0037, -0.0031],
>     [ 0.0038, -0.0031],
>     [ 0.0038, -0.0031],
>     [ 0.0038, -0.0031],
>     [ 0.0037, -0.0029],
>     [ 0.0039, -0.0032],
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/75195557/377287822-5f49d581-0905-4514-9997-030c33a61650.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxMzY4NTksIm5iZiI6MTcyOTEzNjU1OSwicGF0aCI6Ii83NTE5NTU1Ny8zNzcyODc4MjItNWY0OWQ1ODEtMDkwNS00NTE0LTk5OTctMDMwYzMzYTYxNjUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE3VDAzNDIzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBiMTUzYzE5YzYxNGMwYTYzYWFkZDlhZmU0NzRiNGIxN2M3YTM1OWZlOGI4YTY4OGFmMTg3YTRlMTJiYzE3YzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.k30O2vqdO0Q6L_hFirbfsPD1yFI4U5c1FeT_KGyvYmY) ![image](https://private-user-images.githubusercontent.com/75195557/377287858-7c52f7f3-f6e2-4639-acc0-3fe692948b54.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxMzY4NTksIm5iZiI6MTcyOTEzNjU1OSwicGF0aCI6Ii83NTE5NTU1Ny8zNzcyODc4NTgtN2M1MmY3ZjMtZjZlMi00NjM5LWFjYzAtM2ZlNjkyOTQ4YjU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE3VDAzNDIzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhYzc4MWI3YzFkY2NiNmU2MzViNTUwZmE2YTJmMTE3OTA5YzljMzFiYzBkZjMzMTAyZmEwODRhMDc4OWEzZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Gtx6MoG_e-cOxnmjAxBOK5uKNdHmR_1wZtxoxtJ84hs) I guess maybe scaler.scale(loss).backward() make grad too small to updata weight


1.	For the command python main.py -t=DeepFake -d=/mnt/data1/zhangyufei/dataset/FFcored/, since it’s training task1, the distillation loss used in CoReD is not applied. Therefore, during backpropagation, distillation loss and sne loss are not used. You should check that normal training occurs without these losses.
2.	The model obtained from training in step 1 can be tested using the command you provided:
python main.py -s=Face2Face_DeepFake -t=FaceSwap -d=/mnt/data1/zhangyufei/dataset/FFcored/ -w=/mnt/data/zhangyufei/deepfake/CoReD/CoReD_weights.
Here’s my suggestion first:
Please first check that the execution of step 1 is working correctly. This case applies the cross-entropy loss, which is commonly used in supervised classification tasks. If the training is not progressing as expected, please verify that the real and fake data are being loaded properly.

**Comment by alsgkals2** [comment]
> is there differences in --model and --network? xception not need pretrain weight but Efficient need pretrain weight ?

•--network refers to the backbone model used for training, such as **Xception or EfficientNet**, as outlined in the experiment tables of the paper.
•--moodel refers to the method mentioned in the paper, which can include techniques like **knowledge distillation, transfer learning**, or the proposed **CoReD** method.
Thank you.

**Comment by ats4869** [comment]
Nǐ de shùjù jí de wénjiàn géshì shì shénme, wǒ de task1 xiǎnshìle túpiàn shùliàng, wǒ rènwéi shì zhèngquè jiāzàile real hé fake shùjù
49 / 5,000
What is the file format of your dataset? My task1 shows the number of images, which means I have loaded the real and fake data correctly.

**Comment by ats4869** [comment]
What is the file format of your dataset? My task1 shows the number of images, which means I have loaded the real and fake data correctly.

**Comment by ats4869** [comment]
![image](https://github.com/user-attachments/assets/03c8b69b-d13d-4d49-ab5f-3702e2f36511)
